src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+public abstract class ArrayList<T> implements List<T> {
```

Comment:
ArrayList should not be abstract and should implement the List interface properly. Remove the 'abstract' keyword to make it a concrete class.